src/components/TodoForm.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useRef } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  handleSubmit: (event: React.FormEvent) => void;
+  newTitle: string;
+  pending: boolean;
+  setNewTitle: (title: string) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({
+  todos,
+  handleSubmit,
+  newTitle,
```

Comment:
you can move that newTitle state to this component, and just send it as the argument in handleSubmit, it will remove redundant rerenders